core/src/main/java/org/opensearch/sql/expression/aggregation/CalciteOnlyAggregator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.aggregation;
+
+import java.util.List;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.exception.ExpressionEvaluationException;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.function.FunctionName;
+
+/**
+ * Placeholder aggregator for functions that are only supported in Calcite v3 engine. This class
+ * exists to register the function signatures in the legacy v2 system, but will throw an error if
+ * actually executed outside of Calcite.
+ */
+public class CalciteOnlyAggregator extends Aggregator<CalciteOnlyAggregator.CalciteOnlyState> {
```

Comment:
Could you clarify the purpose of this class? I recall we've aligned that V2 support for all ongoing PPL enhancement is not required.

Reply:
This was handling  Removed v2 registration. [fc9123a](https://github.com/opensearch-project/sql/pull/4042/commits/fc9123a0c9c379a77586e43571f8a52b7fef6aaa)